content/administration/upgrade/odoo_sh.rst

```diff
@@ -1,66 +1,132 @@
+   backup and **test it**. Write upgrade scripts if necessary.
+#. Trigger the production upgrade from your :guilabel:`Production` branch and sit tight.
+
+.. seealso::
+   - :doc:`../../administration/upgrade`
+   - :doc:`Upgrade FAQ <../upgrade/faq>`
+   - :doc:`Introduction to Odoo.sh <../odoo_sh/overview/introduction>`
+
+.. _odoo_sh_upgrade/custom-modules:
+
+Upgrade your custom modules
 ===========================
 
-At the beginning of your upgrade project, make sure that you create a new staging branch for testing
-purposes by forking your main branch.
+The first step is to upgrade your custom modules to keep them compatible with the new version. Fork
+your :guilabel:`Production` branch in the :guilabel:`Development` stage, then go to the settings of
+your :guilabel:`Development` branch and select the Odoo version you target. If needed, modify your
+code to be compatible with the new version. Make sure to **test** your features are still working
+correctly.
+
+.. note::
+   Depending on your contract, the upgrade of your custom modules can be done by yourself, by your
+   Partner or by Odoo (if you hold a subscription including maintenance of customizations).
+
+.. _odoo_sh_upgrade/testing-phase:
+
+Upgrade your database on a staging branch
+=========================================
+
+Take the upgraded development branch and drag & drop it to :guilabel:`Staging`.
 
-Once the staging build is complete (it doesn't matter if it failed due to the version
-incompatibility), import your upgraded dump in the *Backups* tab of the branch.
+Go to the :guilabel:`Upgrade` tab and select the :guilabel:`target version`. Then, click on
+:guilabel:`Test Upgrade`.
+
+.. image:: odoo_sh/odoo-sh-staging.png
+   :align: center
+   :alt: Odoo.sh project and tabs
+
+The **latest production daily automatic backup** is sent to the
+`upgrade platform <https://www.upgrade.odoo.com>`_ to start the upgrade test process.
 
 .. note::
-   The platform automatically detects the version of the dump and changes the version of Odoo's
-   source code to the corresponding version of the build.
+   You can follow the upgrade process by going to the :guilabel:`Upgrade` menu of your
+   :guilabel:`Production` branch.
 
-Test the upgraded database and make sure everything runs as it's supposed to.
+When the upgraded backup is ready on the `upgrade platform <https://www.upgrade.odoo.com>`_, it is
+automatically downloaded back to your project.
 
-Upgrade your production database
-================================
+The branch is now in a **special mode**: each time a **commit is pushed** on the branch, a
+**restore operation** of the upgraded backup occurs, and an **update of all the custom modules**
+happens. This allows you to quickly iterate on your custom modules upgrade scripts. The log file of
+the upgrade process can be found at `~/logs/upgrade.log` in your newly upgraded staging build.
```

Comment:
You forgot to apply my previous comment
use the [file markup](https://www.odoo.com/documentation/master/contributing/documentation/rst_cheat_sheet.html#file)

https://github.com/odoo/documentation/pull/1694#discussion_r915843572